pkg/webhooks/handlers.go

```diff
@@ -0,0 +1,225 @@
+	)
+}
+
+func (ws *WebhookServer) admissionHandler(filter bool, inner handlers.AdmissionHandler) http.HandlerFunc {
+	if filter {
+		inner = handlers.Filter(ws.configHandler, inner)
+	}
+	return handlers.Monitor(ws.webhookMonitor, handlers.Admission(ws.log, inner))
+}
+
+func (ws *WebhookServer) policyMutation(request *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
+	logger := setupLogger(ws.log, "policy mutation", request)
+	policy, oldPolicy, err := admissionutils.GetPolicies(request)
+	if err != nil {
+		logger.Error(err, "failed to unmarshal policies from admission request")
+		return admissionutils.ResponseWithMessage(true, fmt.Sprintf("failed to default value, check kyverno controller logs for details: %v", err))
+	}
+	if oldPolicy != nil && isStatusUpdate(oldPolicy, policy) {
+		logger.V(4).Info("skip policy mutation on status update")
+		return admissionutils.Response(true)
+	}
+	startTime := time.Now()
+	logger.V(3).Info("start policy change mutation")
+	defer logger.V(3).Info("finished policy change mutation", "time", time.Since(startTime).String())
+	// Generate JSON Patches for defaults
+	if patches, updateMsgs := policymutation.GenerateJSONPatchesForDefaults(policy, logger); len(patches) != 0 {
+		return admissionutils.ResponseWithMessageAndPatch(true, strings.Join(updateMsgs, "'"), patches)
+	}
+	return admissionutils.Response(true)
+}
+
+//policyValidation performs the validation check on policy resource
+func (ws *WebhookServer) policyValidation(request *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
+	logger := setupLogger(ws.log, "policy validation", request)
+	policy, oldPolicy, err := admissionutils.GetPolicies(request)
+	if err != nil {
+		logger.Error(err, "failed to unmarshal policies from admission request")
+		return admissionutils.ResponseWithMessage(true, fmt.Sprintf("failed to validate policy, check kyverno controller logs for details: %v", err))
+	}
+	if oldPolicy != nil && isStatusUpdate(oldPolicy, policy) {
+		logger.V(4).Info("skip policy validation on status update")
+		return admissionutils.Response(true)
+	}
+	startTime := time.Now()
+	logger.V(3).Info("start policy change validation")
+	defer logger.V(3).Info("finished policy change validation", "time", time.Since(startTime).String())
+	response, err := policyvalidate.Validate(policy, ws.client, false, ws.openAPIController)
+	if err != nil {
+		logger.Error(err, "policy validation errors")
+		return admissionutils.ResponseWithMessage(true, err.Error())
+	}
+	if response != nil && len(response.Warnings) != 0 {
+		return response
+	}
+	return admissionutils.Response(true)
+}
+
+// resourceMutation mutates resource
+func (ws *WebhookServer) resourceMutation(request *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
+	logger := setupLogger(ws.log, "resource mutation", request)
+	if excludeKyvernoResources(request.Kind.Kind) {
+		return admissionutils.ResponseSuccess(true, "")
+	}
+	if request.Operation == v1beta1.Delete {
```

Comment:
It looks like we no longer register DELETE operation in the mutating webhook: https://github.com/kyverno/kyverno/blob/9fc65fa5a7bc2721d1d5e79fd0ea49718ed264c9/pkg/webhookconfig/resource.go#L78

@prateekpandey14 - can you please confirm and remove it?